tools/maf_clean_up.py

```diff
@@ -0,0 +1,177 @@
+input_maf = open ("data_mutations_extended.txt", "r")
+header_line = "";
+
+# search for header row
+for line in input_maf:
+	values = line.split("\t");
+	if values[0].upper() == "HUGO_SYMBOL":
+		header_line = line;
+
+headers = header_line.rstrip().split("\t");
+keep_col_index_arr = []; #Index of columns to keep
+input_maf.close();
+
+# iterate through all column headers and record the index of targeted ones
+count = 0;
+exon_number_index = -1;
+exon_index = -1;
+intron_index = -1;
+oncotator_index_arr = [];
+for header_item in headers:
+	if header_item.upper() == "EXON_NUMBER":
+		exon_number_index = count;
+	elif header_item.upper() == "EXON":
+		exon_index = count;
+	elif header_item.upper() == "INTRON":
+		intron_index = count;
+	elif header_item[0:len("ONCOTATOR")] == "ONCOTATOR":
+		oncotator_index_arr.append(count);
+	else:
+		keep_col_index_arr.append(count);
+	count += 1;
+
+print "INDEX OF COLUMNS TO DROP >>>>>>>>>>>>>>";
+print "INTRON: " + str(intron_index);
+print "EXON_NUMBER: " + str(exon_number_index);
+print "EXON: " + str(exon_index);
+print "ONCOTATOR: " + str(oncotator_index_arr)[1:-1] ;
+print "DROPPING COLUMNS ....";
+
+
+#### Start of reconstructing file
+input_maf = open ("data_mutations_extended.txt", "r");
+output_maf = open ("output_maf.txt", "w");
+
+# write out the original extra header lines (sequences samples, versions, etc.)
+for line in input_maf:
+	values = line.split("\t");
+	if values[0].upper() != "HUGO_SYMBOL":
+		output_maf.write(line);
+	else:
+		break;
+
+# write out header line
+for keep_col_index in keep_col_index_arr:
+	if headers[keep_col_index].upper() == "HUGO_SYMBOL":
+		output_maf.write("Hugo_Symbol");
+	elif headers[keep_col_index].upper() == "START_POSITION":
+		output_maf.write("\tStart_Position");
+	elif headers[keep_col_index].upper() == "END_POSITION":
+		output_maf.write("\tEnd_Position");
+	else:
+		output_maf.write("\t" + headers[keep_col_index]);
+output_maf.write("\n");
+
+# write out content
+for line in input_maf:
+	col_counter = 0;
+	cols = line.split("\t");
+	for keep_col_index in keep_col_index_arr:
+		if col_counter != 0:
+			output_maf.write("\t");
+			output_maf.write(cols[keep_col_index]);
+		else:
```

Comment:
Not necessary, but this part could also be written as a dictionary stored as key-value pairs. Then this writing portion can be done through a function that checks whether or not the value exists in the dictionary and writes out the corresponding value.